add: A minimum implementation of prompt()

gugod opened the pull request:
As a way to address the issue of #843

gugod committed:
add: A minimum implementation of prompt()

gugod committed:
Replace the use of ExtUtils::MakeMaker::prompt() is our own prompt().

gugod committed:
Merge branch 'develop' into prompt-helper

guest20 commented on lib/App/Perlbrew/Util.pm:
```diff
@@ -151,4 +152,21 @@ sub make_skaji_relocatable_perl_url {
     return undef;
 }
 
+sub prompt {
```
Is it also worth checking `PERL_MM_USE_DEFAULT` too?

gugod replied:
@guest20 Nice idea. That would make perlbrew backward compatible and at least we could support that env var for a few more versions. Thanks for the comment!

guest20 replied:
_Change is evil, unless it's the kind of change that leaves things exactly as they were_

gugod committed:
add: Support of PERL_MM_USE_DEFAULT in prompt().

Since perlbrew has been using ExtUtils::MakeMaker::prompt() since the
beginning, PERL_MM_USE_DEFAULT has been indirectly supported as a
mean. We might as well keep supporting it, at least for a few more
versions, to maintain the backward compatibility.

gugod commented:
Probably want to research on the behavior when STDIN is not a tty (non-interactive shell) and when STDIN itself never has any input. That is, in shell scripts, the following two statements should behave the same (using `install-cpm` as an example because it calls prompt() internally):

    # 1. STDIN has a "\n". The default choice should be made.
    echo | perlbrew install-cpm
    
    # 2. STDIN has nothing at all. The default choice should be made.
    perlbrew install-cpm

gugod committed:
Merge branch 'develop' into prompt-helper

gugod committed:
`mbtiny regenerate`

gugod committed:
test: cleanup side-effects.
